### ISSUE TITLE
Support `pnpm` workspaces

### ISSUE THREAD

**Issue by tjmgregory** [ghiscoding/lerna-lite#100]
### Motivation / Use Case
I would like a repo that uses [pnpm workspaces](https://pnpm.io/workspaces) to be able to use `lerna-lite`.

### Expected Behavior
When `version` or `publish` are run in repositories that use [pnpm workspaces](https://pnpm.io/workspaces), the commands should work.

Currently, the following error occurs:

```sh
~/elephant/web-packages -> chore/pnpm |• $
❯ pnpm lerna publish
lerna notice cli v1.0.5
lerna info versioning independent
lerna ERR! Error: Unsupported URL Type "workspace:": workspace:^0.3.1
lerna ERR!     at unsupportedURLType (/Users/theo/elephant/web-packages/node_modules/.pnpm/npm-package-arg@9.0.2/node_modules/npm-package-arg/lib/npa.js:262:15)
lerna ERR!     at fromURL (/Users/theo/elephant/web-packages/node_modules/.pnpm/npm-package-arg@9.0.2/node_modules/npm-package-arg/lib/npa.js:322:13)
lerna ERR!     at Function.resolve (/Users/theo/elephant/web-packages/node_modules/.pnpm/npm-package-arg@9.0.2/node_modules/npm-package-arg/lib/npa.js:80:12)
lerna ERR!     at /Users/theo/elephant/web-packages/node_modules/.pnpm/@lerna-lite+core@1.0.5/node_modules/@lerna-lite/core/dist/package-graph/package-graph.js:53:60
lerna ERR!     at Array.forEach (<anonymous>)
lerna ERR!     at /Users/theo/elephant/web-packages/node_modules/.pnpm/@lerna-lite+core@1.0.5/node_modules/@lerna-lite/core/dist/package-graph/package-graph.js:47:44
lerna ERR!     at Map.forEach (<anonymous>)
lerna ERR!     at new PackageGraph (/Users/theo/elephant/web-packages/node_modules/.pnpm/@lerna-lite+core@1.0.5/node_modules/@lerna-lite/core/dist/package-graph/package-graph.js:43:14)
lerna ERR! lerna Unsupported URL Type "workspace:": workspace:^0.3.1
```

### Other Information

- This same problem is present in the forked lerna repository https://github.com/lerna/lerna/issues/2564#issuecomment-1035392832
- I don't mind taking a crack at this myself when I can find the time 👍 
- My focus is `pnpm` as that is my personal use-case, however this issue is present with `yarn` workspaces also.

**Comment by ghiscoding** [comment]
I don't use pnpm so yeah I cannot help on this problem and even though I created this fork I don't know the code well enough to work on it either. However I remember there was some discussions around that subject in Lerna with multiple issues and maybe this opened [PR #2450](https://github.com/lerna/lerna/pull/2450) and perhaps it's probably better to read this [comment](https://github.com/lerna/lerna/pull/2450#issuecomment-854001468) from Lerna's author and use that as a base.

**Comment by tjmgregory** [comment]
@ghiscoding Amazing stuff, thanks for the link I'd not come that yet. Will let you know if I manage to give it a crack 🤘

**Comment by tjmgregory** [comment]
In the end we've just stripped out use of the `workspace:` protocol. Not ideal, but consequently will unlikely find time in the near term to have a crack at this - apologies 🙏

**Comment by ghiscoding** [comment]
no worries, anyone who uses this feature (I'm not) and has time to take it over then feel free to contribute at any time.

**Comment by ghiscoding** [comment]
@tjmgregory 
I got a potential Beta on this feature if you or anyone else want to give me a hand to test with a real project (which I do not have). This `workspace:` protocol is totally new to me and so if I could get help from you or anyone else, that would be great and I could provide the steps to test it out, something that would look like these steps
1. build the project TS to JS
2. npm link it to your local
3. test the `version` command in your project with `git-dry-run` to inspect changes
4. test the `publish` command...
5. test the `version` for real
6. test the `publish` for real

even though this protocol is totally new to me, I think I got most of everything covered. 

If anyone want to help please reach out, and/or if anyone could review the PR with the changes and possibly provide feedback

Cheers

**Comment by ghiscoding** [comment]
released under [1.2.0](https://github.com/ghiscoding/lerna-lite/releases/tag/v1.2.0)

**Comment by ghiscoding** [comment]
hey guys, I actually released [1.3.0](https://github.com/ghiscoding/lerna-lite/releases/tag/v1.3.0) to make it more strict match, see the [release 1.3.0](https://github.com/ghiscoding/lerna-lite/releases/tag/v1.3.0) for more info. I strongly suggest to use the latest version. 

Cheers

**Comment by tjmgregory** [comment]
Heya @ghiscoding - thanks for all your _incredible_ work. We're all setup on 1.3.0 and everything is working like a charm ✨ great work!
